docs/middleware.rst

```diff
@@ -1,34 +1,34 @@
+.. _internals__middlewares:
+
 Middleware
 ==========
 
-Web3 manages layers of middlewares by default. They sit between the public Web3 methods and the
-:doc:`providers`, which handle native communication with the Ethereum client. Each layer
-can modify the request and/or response. Some middlewares are enabled by default, and
-others are available for optional use.
+Web3 is instantiated with layers of middleware by default. They sit between the public
```

Comment:
If reference to the class let's either link to the `py:class` definition in the docs or use a code block here. It helps clarify we are talking about the class.